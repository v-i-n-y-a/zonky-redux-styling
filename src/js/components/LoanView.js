import React, { Component } from "react";
import moment from "moment";
import Img from "./Img";

const Info = ({ title, value }) => {
  if (value === true) value = "Ano";
  else if (value === false) value = "Ne";
  return (
    <p class="info">
      <span class="info-title">{title}:</span>
      <span class="info-value">{value}</span>
    </p>
  );
};

const LoanView = ({ loan }) => {
  return (
    <div class="loan-view jumbotron list-group">
      <div class="row">
        <div class="col-md-4">
          {loan.photos.map((photo, index) => (
            <Img key={index} src={photo.url} alt={photo.name} />
          ))}

          <h2>Další informace</h2>
          <Info title="Úroková sazba" value={`${loan.interestRate}%`} />
          <Info
            title="Uzávěrka"
            value={moment(loan.deadline).format("YYYY-MM-DD")}
          />
          <Info title="Částka" value={`${loan.amount}Kč`} />
          <Info title="Na jak dlouho" value={`${loan.termInMonths} Měsicu`} />
        </div>
        <div class="col-md-6">
          <h1> {loan.name} </h1>
          <div> {loan.story} </div>
          <div class="loan-view-details">
            <Info title="Autor" value={loan.nickName} />
            <Info
              title="Bylo Publikováno"
              value={moment(loan.datePublished).format("YYYY-MM-DD")}
            />
            <Info title="Hodnocení" value={loan.rating} />
          </div>
        </div>
      </div>
    </div>
  );
};

export default LoanView;
