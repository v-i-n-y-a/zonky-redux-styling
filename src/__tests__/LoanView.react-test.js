import React from 'react';
import LoanView from '../js/components/LoanView.js';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import loanJSON from './loanJSON';

describe('<LoanView loan={loan} />', () => {
  it('LoanView renders correctly', () => {
    const component = renderer.create(<LoanView loan={loanJSON} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
