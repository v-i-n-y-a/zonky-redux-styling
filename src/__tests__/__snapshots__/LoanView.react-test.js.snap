// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoanView loan={loan} /> LoanView renders correctly 1`] = `
<div
  class="loan-view jumbotron list-group"
>
  <div
    class="row"
  >
    <div
      class="col-md-4"
    >
      <img
        alt="file.png"
        class="img-responsive"
        src="/loans/marketplace/loans/88873/photos/8937"
      />
      <h2>
        Další informace
      </h2>
      <p
        class="info"
      >
        <span
          class="info-title"
        >
          Úroková sazba
          :
        </span>
        <span
          class="info-value"
        >
          0.0499%
        </span>
      </p>
      <p
        class="info"
      >
        <span
          class="info-title"
        >
          Uzávěrka
          :
        </span>
        <span
          class="info-value"
        >
          2017-05-16
        </span>
      </p>
      <p
        class="info"
      >
        <span
          class="info-title"
        >
          Částka
          :
        </span>
        <span
          class="info-value"
        >
          350000Kč
        </span>
      </p>
      <p
        class="info"
      >
        <span
          class="info-title"
        >
          Na jak dlouho
          :
        </span>
        <span
          class="info-value"
        >
          38 Měsicu
        </span>
      </p>
    </div>
    <div
      class="col-md-6"
    >
      <h1>
         
        Pomoc investorů investorovi
         
      </h1>
      <div>
         
        Mala hystorie
         
      </div>
      <div
        class="loan-view-details"
      >
        <p
          class="info"
        >
          <span
            class="info-title"
          >
            Autor
            :
          </span>
          <span
            class="info-value"
          >
            zonky114025
          </span>
        </p>
        <p
          class="info"
        >
          <span
            class="info-title"
          >
            Bylo Publikováno
            :
          </span>
          <span
            class="info-value"
          >
            2017-05-14
          </span>
        </p>
        <p
          class="info"
        >
          <span
            class="info-title"
          >
            Hodnocení
            :
          </span>
          <span
            class="info-value"
          >
            AAAA
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<loanJSON object for testing  /> it it will be changed Layout and ListView test will be changed also 1`] = `
Object {
  "amount": 350000,
  "covered": false,
  "datePublished": "2017-05-14T17:09:54.768+02:00",
  "deadline": "2017-05-16T16:50:50.922+02:00",
  "id": 1,
  "interestRate": 0.0499,
  "investmentRate": 0.33085714285714285,
  "investmentsCount": 229,
  "mainIncomeType": "SELF_EMPLOYMENT",
  "name": "Pomoc investorů investorovi",
  "nickName": "zonky114025",
  "photos": Array [
    Object {
      "name": "file.png",
      "url": "/loans/88873/photos/8937",
    },
  ],
  "published": true,
  "purpose": "8",
  "questionsCount": 1,
  "rating": "AAAA",
  "region": "14",
  "remainingInvestment": 234200,
  "story": "Mala hystorie",
  "termInMonths": 38,
  "topped": null,
  "userId": 114025,
}
`;
